sympy/solvers/tests/test_solvers.py

```diff
@@ -114,6 +114,446 @@ def test_guess_transcendental():
+    e = lx
+    assert [solve(e, **f) for f in flags] == \
+        [[0],
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve([e], **f) for f in flags] == [
+        {x: 0},
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve(e, x, **f) for f in flags] == \
+        [[0],
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve([e], x, **f) for f in flags] == [
+        {x: 0},
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve(e, {x}, **f) for f in flags] == \
+        [[0],
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve([e], {x}, **f) for f in flags] == [
+        {x: 0},
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve(e, y, x, **f) for f in flags] == \
+        [[(y, 0)],
+        [{x: 0}],
+        ([y, x], {(y, 0)})]
+        #([x, y], {(0, y)})]master
+    assert [solve([e], y, x, **f) for f in flags] == [
+        {x: 0},
+        [{x: 0}],
+        ([y, x], {(y, 0)})]
+        #([x], {(0,)})]#master
+    assert [solve(e, {y, x}, **f) for f in flags] == \
+        [[{x: 0}],
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve([e], {y, x}, **f) for f in flags] == [
+        #[{x: 0}],
+        {x: 0},#master
+        [{x: 0}],
+        ([x], {(0,)})]
+    assert [solve(e, y, **f) for f in flags] == \
+        [[],
+        [],
+        ([y], set())]
+    assert [solve([e], y, **f) for f in flags] == [
+        [],
+        [],
+        ([y], set())]
+    e = nx
+    def lot(lod):
+        from sympy.core.sorting import ordered
+        s = list(ordered(lod[0].keys()))
+        return [tuple([d[i] for i in s]) for d in lod]
+    assert [solve(e, **f) for f in flags] == \
+        [[-2, 2],
+        [{x: -2}, {x: 2}],
+        ([x], {(2,), (-2,)})]
+    assert [solve([e], **f) for f in flags] == [
+        [{x: -2}, {x: 2}],
+        [{x: -2}, {x: 2}],
+        ([x], {(2,), (-2,)})]
+    assert [solve(e, x, **f) for f in flags] == \
+        [[-2, 2],
+        [{x: -2}, {x: 2}],
+        ([x], {(2,), (-2,)})]
+    assert [solve([e], x, **f) for f in flags] == [
+        lot([{x: -2}, {x: 2}]),
+        [{x: -2}, {x: 2}],
+        ([x], {(2,), (-2,)})]
+    assert [solve(e, {x}, **f) for f in flags] == \
+        [[-2, 2],
+        [{x: -2}, {x: 2}],
+        ([x], {(2,), (-2,)})]
+    assert [solve([e], {x}, **f) for f in flags] == [
+        [{x: -2}, {x: 2}],
+        [{x: -2}, {x: 2}],
+        ([x], {(2,), (-2,)})]
+    assert [solve(e, x, y, **f) for f in flags] == \
+        [[(-2, y), (2, y)],
+        [{x: -2}, {x: 2}],
+        ([x, y], {(2, y), (-2, y)})]
+    assert [solve([e], x, y, **f) for f in flags] == [
+        [(-2, y), (2, y)],
+        [{x: -2}, {x: 2}],
+        ([x, y], {(2, y), (-2, y)})]
+    assert [solve(e, y, x, **f) for f in flags] == \
+        [[(y, -2), (y, 2)],
+        [{x: -2}, {x: 2}],
+        ([y, x], {(y, -2), (y, 2)})]
+    assert [solve([e], y, x, **f) for f in flags] == [
+        [(y, -2), (y, 2)],
+        [{x: -2}, {x: 2}],
```

Comment:
Why are these two different?

Reply:
It's hard to tell from context what you are referring to. Are you asking why tuples are used at L219 and dictionaries at L220? Or are you asking why specifying the order `y, x` in L218 gives different results from using a set at L221? The latter is because if the order is indeterminate, the output cannot be used unambiguously to know which values correspond with which variables unless the ordering routine is known (and cf #23828 for how arbitrary that can be).

Reply:
I think I highlighted the wrong examples (it's quite confusing to see which example is being tested where in this function). We have:
```python
In [1]: solve([x**2 - 4], y, x)
Out[1]: [(y, -2), (y, 2)]

In [2]: solve([x**2 - 4], y, x, set=True)
Out[2]: ([x, y], {(-2, y), (2, y)})
```
With `set=True` here a symbol ordering `[y, x]` has been provided through the ordering of the unknowns in `*args` but the example comes back using a different ordering `[x, y]`.

Reply:
The set form is canonical regardless of how you pass the symbols, so `...x, y, set=True` = `...y,x,set=True` = `...{x,y}, set=True`. It is not hard, however, to keep the order if the order is determinate since, without adding a new flag `tuple=True` that is a way to get full ordered tuples back from solve.

Reply:
That can be handled in a separate PR. The purpose of this is to

1. capture the solve io fingerprint so subsequent mods will flag any differences and to identify issues that should be resolved (like the case here that you identified)
2. raise error on duplicate symbols
3. apply simplify per request on a case that was being missed
4. correct output for set=True on the corner case identified